### ISSUE TITLE
Turkish Character in url

### ISSUE THREAD

**Issue by Selimerkan** [q2a/question2answer#348]
Hello
Not changing special character to latin characters. And I think it is bad for seo.
Example
http://www.kampusgenc.com/311/hutin-anla%C5%9Fmas%C4%B1-%C3%B6nemi-nedir
Ö-o Ü-u -ı -i

**Comment by svivian** [comment]
Thanks for reporting, I will check into this. I did see your Q2A post but didn't get around to responding, sorry.

**Comment by svivian** [comment]
Hi @Selimerkan I've just tested this and it's working fine for me. The characters in your URL are already in the `$qa_utf8removeaccents` variable in `util/string.php`

Do you have the option "Remove accents from question URLs" checked in admin?

**Comment by DanielRuf** [comment]
I think there are also issues with some other languages and character sets (cyrillic, asian, ...).

`$string="äüöß ľ š č ť ž ý á í é Č Á Ž Ý ZacarÃƒÂ­as FerreÃƒÂ­ra qqqqŻŹĆŃĄŚŁĘÓżźćńąśłęóqqq A æ Übérmensch på høyeste nivå! И я люблю PHP! есть. ﬁ ¦ 电脑坏了。";`

http://ideone.com/K9Fk2m

I tried different solutions and most do not work or need characters manually added as they do not contain all special characters. And Transliterator is not available by default.

The best result gave me https://github.com/jbroadway/urlify

```php
include 'vendor/autoload.php';
$string="äüöß ľ š č ť ž ý á € í é Č Á Ž Ý ZacarÃƒÂ­as FerreÃƒÂ­ra qqqqŻŹĆŃĄŚŁĘÓżźćńąśłęóqqq A æ Übérmensch på høyeste nivå! И я люблю PHP! есть. ﬁ ¦ 电脑坏了。";
echo URLify::filter ($string, mb_strlen($string));
// auoss-s-c-t-z-y-i-e-c-z-y-zacara-ferreara-qqqqzzcnasleozzcnasleoqqq-ae-ubermensch-pa-hoyeste-niva-i-ya-lyublyu-php-est
```

URLs should just contain ASCII characters and also just in lowercase.

URLify might be useful.

**Comment by svivian** [comment]
Fixed in 132973eaaedbb19f3ef578f35a4291d177a3a370 - we now remove all non-latin characters that weren't converted by `qa_string_remove_accents`.

This may not be entirely what you wanted as Turkish characters are not converted. I thought about adding all the possibilities like URLify does, but I think it's best solved with an override plugin that updates the `$qa_utf8removeaccents` variable for your locale (by overriding the `qa_string_initialize` function).
